test/LaunchDarkly.EventSource.Tests/EventSourceTests.cs

```diff
@@ -192,6 +192,39 @@ public async Task When_LastEventId_is_configured_then_the_outgoing_request_conta
             Assert.True(request.Headers.GetValues(Constants.LastEventIdHttpHeader).Contains(lastEventId));
         }
 
+        [Fact]
+        public async Task When_reconnecting_the_outgoing_request_contains_Last_Event_Id_header()
+        {
+            var lastEventId = "10";
+            var firstResponse = $"id:{lastEventId}\nevent: put\ndata: this is a test message\n\n";
+            var secondResponse = $"id:20\nevent: put\ndata: this is a test message\n\n";
+
+            var handler = new StubMessageHandler();
+            handler.QueueResponse(StubResponse.StartStream(StreamAction.Write(firstResponse), StreamAction.CloseStream()));
+            handler.QueueResponse(StubResponse.StartStream(StreamAction.Write(secondResponse), StreamAction.CloseStream()));
+
+            var evt = new EventSource(new Configuration(_uri, handler));
+            var first = true;
+            handler.RequestReceived += (s, r) =>
+            {
+                if (first)
+                {
+                    first = false;
+                }
+                else
+                {
+                    evt.Close();
+                }
+            };
+
+            await evt.StartAsync();
+
+            var requests = handler.GetRequests().ToList();
+            Assert.False(requests[0].Headers.Contains(Constants.LastEventIdHttpHeader));
+            Assert.True(requests[1].Headers.Contains(Constants.LastEventIdHttpHeader));
+            Assert.True(requests[1].Headers.GetValues(Constants.LastEventIdHttpHeader).Contains(lastEventId));
+        }
+
```

Comment:
As per the previous comment, I would add a second version of this test in which the original configuration specifies a different last event ID.

Reply:
I added a test where it sets both the `LastEventId` configuration option and manually adds a `Last-Event-Id` header. The behaviour you mentioned was only triggered when adding the header.